pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -366,16 +375,18 @@ func (service *SecurityPolicyService) buildExpressionsMatchExpression(matchExpre
 
 // build appliedTo group ID for both policy and rule levels.
 func (service *SecurityPolicyService) buildAppliedGroupID(obj *v1alpha1.SecurityPolicy, ruleIdx int, createdFor string) string {
-	prefix := common.SecurityPolicyPrefix
-	if createdFor == common.ResourceTypeNetworkPolicy {
-		prefix = common.NetworkPolicyPrefix
-	}
-
 	ruleIdxStr := ""
 	if ruleIdx != -1 {
 		ruleIdxStr = fmt.Sprintf("%d", ruleIdx)
 	}
+	if isVpcEnabled(service) {
+		return util.GenerateID(util.GenerateIDByObject(obj), "", common.TargetGroupSuffix, ruleIdxStr)
```

Comment:
suffix = common.TargetGroupSuffix
if ruleIdx != -1 {
    suffix = fmt.Sprintf("%d_%s", ruleIdx, suffix)
}

return util.GenerateIDByObjectWithSuffix(obj, suffix)

Keep using GenerateIDByObjectWithSuffix for suffix case like buildRulePeerGroupID,   is it better?

Reply:
updated.